### PR TITLE
Registration cache

### PR DESCRIPTION
**Pull request by lukanus**
# What 🕵️‍♀️
Registration cache for validator registrations

# Why 🔑
To limit the number of registration updates

### CODE REVIEW

File: pkg/relay/processmanager.go [hunk 1]
```diff
@@ -37,11 +41,16 @@ type ProcessManager struct {
 	m ProcessManagerMetrics
 }
 
-func NewProcessManager(l log.Logger, verifySize, storeSize uint) *ProcessManager {
-	rm := &ProcessManager{
-		l:           l,
-		LastRegTime: make(map[string]uint64),
+func NewProcessManager(l log.Logger, verifySize, storeSize uint, registrationCacheSize int) *ProcessManager {
```
Comment (aratz-lasa) [inline pkg/relay/processmanager.go:44]: Why don't we return `(*ProcessManager, error)`?

**Review by aratz-lasa — COMMENTED** [review]
I just have one comment, where we should return the struct and the error, instead of returning nil on error.

File: pkg/relay/processmanager.go (hunk 1 above)
Comment (lukanus) [reply pkg/relay/processmanager.go:44]: In general constructors shouldn't return an error ;) but yeah... you're right, I'll change that

**Review by aratz-lasa — APPROVED** [review]
This is an amazing idea @lukanus, amazing work!

**Review by aratz-lasa — APPROVED** [review]
Just a small comment to remove a TODO, but LTGMT

**Pull request merged**